pages/settings/wallets/ConnectWallet.tsx

```diff
@@ -0,0 +1,508 @@
+    "get_info",
+    "get_balance",
+    "get_budget",
+    "make_invoice",
+    "pay_invoice",
+    "lookup_invoice",
+    "list_transactions",
+    "sign_message",
+  ];
+  const maxAmount = options.maxAmount || 100_000_000;
+  const satsAmount = maxAmount / 1000;
+  const { data: info } = useInfo();
+
+  const [isCreatingConnection, setCreatingConnection] = React.useState(false);
+  const wallets = useAppStore((store) => store.wallets);
+  const selectedWalletId = useAppStore((store) => store.selectedWalletId);
+
+  const [deeplinkConnectionSecret, setDeeplinkConnectionSecret] =
+    React.useState<string | null>(null);
+  const [redirectCountdown, setRedirectCountdown] = React.useState<number>();
+  const [connectionCreated, setConnectionCreated] = React.useState(false);
+
+  React.useEffect(() => {
+    if (redirectCountdown === undefined) {
+      return;
+    }
+
+    if (redirectCountdown === 0) {
+      (async () => {
+        if (returnTo) {
+          let url = new URL(returnTo);
+          if (deeplinkConnectionSecret) {
+            url.searchParams.set("value", deeplinkConnectionSecret);
+          }
+          try {
+            console.info("opening URL", url.toString());
+            await openURL(url.toString());
+          } catch (error) {
+            console.error(error);
+            errorToast(
+              new Error("Couldn't open URL, do you have the app installed?"),
+            );
+          }
+        }
+        if (router.canDismiss()) {
+          router.dismissAll();
+        }
+        router.replace("/");
+      })();
+      return;
+    }
+
+    const timer = setTimeout(() => {
+      setRedirectCountdown((prev) => (prev ? prev - 1 : prev));
+    }, 1000);
+    return () => clearTimeout(timer);
+  }, [redirectCountdown, deeplinkConnectionSecret, returnTo]);
+
+  const confirm = async () => {
+    setCreatingConnection(true);
+    try {
+      const nwcClient = useAppStore.getState().nwcClient;
+      if (!nwcClient) {
+        throw new Error("NWC client not connected");
+      }
+
+      let secretKey: string | undefined;
+      if (!pubkey) {
+        const secretKeyBytes = generateSecretKey();
+        secretKey = bytesToHex(secretKeyBytes);
+        pubkey = getPublicKey(secretKeyBytes);
+      }
+
+      const response = await nwcClient.createConnection({
+        pubkey,
+        name: name as string,
+        max_amount: maxAmount,
+        budget_renewal: budgetRenewal,
+        request_methods: requestMethods,
+        notification_types: notificationTypes,
+        expires_at: expiresAt,
+        isolated,
+        metadata,
+      });
+
+      console.info("createConnection response", response);
+
+      if (flow === "deeplink") {
+        setDeeplinkConnectionSecret(
+          `nostr+walletconnect://${response.wallet_pubkey}?secret=${
+            secretKey
+          }&relay=${nwcClient.relayUrl}`,
+        );
+      }
+      setRedirectCountdown(5);
```

Comment:
Here we artificially delay the flow by 5 seconds, I wonder if it's too much. What is the reasoning here?

Reply:
I noticed it was a bit long. Maybe 3 seconds should be good I think, and then let's see feedback from other people. I think the tick is good to have so the user knows the connection succeeds (especially if there is no redirect url set)

Reply:
Yeah 3 should be good, changed!